### PR TITLE
validate: add option to omit unused keys warnings

### PR DESCRIPTION
**Pull request by ajeddeloh**

### CODE REVIEW

**Comment by ajeddeloh** [comment]
It's needed for https://github.com/coreos/container-linux-config-transpiler/pull/57#pullrequestreview-33109317 (I'll update the commit message). That clct PR changes the ValidateWithoutSource that gets called on the generated ignition structure to a plain Validate() using a modified version of the clct config parse tree. The problem is clct has a bunch of options that ignition does not (e.g. etcd) that generate unused key warnings.

**Comment by ajeddeloh** [comment]
Updated to have a better commit message

**Comment by dghubble** [comment]
Specifically, your aim is that all users of `Parse` will still (always) see unused key (fields) warnings, but CT has the flexibility to invoke `Validate` directly and disable unused keys warning. Is there a specific example you have in mind? I haven't run into this and the linked issue upstream doesn't really complain about it specifically.

Please rebase once more to update this branch since its out of date with base.

**Comment by ajeddeloh** [comment]
In the linked PR (I accidentally linked to a section of feedback instead of the full PR), it's used when calling Validate() on the generated ignition configuration with the parse tree from the clc: https://github.com/coreos/container-linux-config-transpiler/pull/57/commits/14f0440a3fde2ae493d5f30424b441d5ef5a4694#diff-8f256984855e8444054b0ecb38776f8eR60

**Comment by dghubble** [comment]
LGTM. The way this repo is setup to require all branches be update to date, you'll wanna rebase/update again.

**Comment by ajeddeloh** [comment]
Rebased.

File: config/validate/validate.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ type AstNode interface {
 
 // Validate walks down a struct tree calling Validate on every node that implements it, building
 // A report of all the errors, warnings, info, and deprecations it encounters
-func Validate(vObj reflect.Value, ast AstNode, source io.ReadSeeker) (r report.Report) {
+func Validate(vObj reflect.Value, ast AstNode, source io.ReadSeeker, checkUnusedKeys bool) (r report.Report) {
```
Comment (ajeddeloh) [inline config/validate/validate.go]: Fixed.

**Pull request merged**